testcases/OpTestSensors.py

```diff
@@ -0,0 +1,171 @@
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+#  @package OpTestSensors
+#  Sensors package for OpenPower testing.
+#
+#  This class will test the functionality of following drivers
+#  1. Hardware monitoring sensors(hwmon driver) using sensors utility
+
+import time
+import subprocess
+import re
+
+from common.OpTestBMC import OpTestBMC
+from common.OpTestIPMI import OpTestIPMI
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.OpTestError import OpTestError
+from common.OpTestLpar import OpTestLpar
+from common.OpTestUtil import OpTestUtil
+
+
+class OpTestSensors():
+    ##  Initialize this object
+    #  @param i_bmcIP The IP address of the BMC
+    #  @param i_bmcUser The userid to log into the BMC with
+    #  @param i_bmcPasswd The password of the userid to log into the BMC with
+    #  @param i_bmcUserIpmi The userid to issue the BMC IPMI commands with
+    #  @param i_bmcPasswdIpmi The password of BMC IPMI userid
+    #  @param i_ffdcDir Optional param to indicate where to write FFDC
+    #
+    # "Only required for inband tests" else Default = None
+    # @param i_lparIP The IP address of the LPAR
+    # @param i_lparuser The userid to log into the LPAR
+    # @param i_lparPasswd The password of the userid to log into the LPAR with
+    #
+    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
+                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_lparip=None,
+                 i_lparuser=None, i_lparPasswd=None):
+        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
+        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
+                                  i_ffdcDir)
+        self.cv_LPAR = OpTestLpar(i_lparip, i_lparuser, i_lparPasswd)
+        self.util = OpTestUtil()
+
+    ##
+    # @brief This function will cover following test steps
+    #        1. It will check for kernel config option CONFIG_SENSORS_IBMPOWERNV
+    #        2. It will load ibmpowernv driver only on powernv platform
+    #        3. It will check for sensors command existence and lm_sensors package
+    #        4. start the lm_sensors service and detect any sensor chips
+    #           using sensors-detect.
+    #        5. At the end it will test sensors command functionality
+    #           with different options
+    #
+    # @param
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def test_hwmon_driver(self):
+
+        # Get OS level
+        l_oslevel = self.cv_LPAR.lpar_get_OS_Level()
+
+        # Get kernel version
+        l_kernel = self.cv_LPAR.lpar_get_kernel_version()
+
+        # Checking for sensors config option CONFIG_SENSORS_IBMPOWERNV
+        cmd = "cat /boot/config-%s | grep -i SENSORS_IBMPOWERNV" % l_kernel
+        print cmd
+        try:
+            res = self.cv_LPAR._ssh_execute(cmd)
```

Comment:
The _ssh_execute is supposed to be a private interface.  Should create a helper interface (i.e. cv_LPAR.lpar_check_config(SENSORS_IBMPOWERNV)).